casbin/model/evaluator.cpp

```diff
@@ -54,6 +54,7 @@ void ExprtkEvaluator::LoadFunctions() {
     AddFunction("keyMatch", ExprtkFunctionFactory::GetExprtkFunction(ExprtkFunctionType::KeyMatch, 2));
     AddFunction("keyMatch2", ExprtkFunctionFactory::GetExprtkFunction(ExprtkFunctionType::KeyMatch2, 2));
     AddFunction("keyMatch3", ExprtkFunctionFactory::GetExprtkFunction(ExprtkFunctionType::KeyMatch3, 2));
+    AddFunction("keyMatch4", ExprtkFunctionFactory::GetExprtkFunction(ExprtkFunctionType::KeyMatch4, 2));
```

Comment:
Do we need to register `keyGet*` as well?

Reply:
Golang Casbin doesn't introduce modules like `exprtk` to parse `keyMatch*` and `keyGet*` expressions for unit testing, but directly determines whether the return value of the function is equal to the expected result. 
At the same time, since the structure of [`ExprtkOtherFunction`](https://github.com/casbin/casbin-cpp/blob/master/include/casbin/model/exprtk_config.h#L110) designed in the previous code only supports the parsing of `*Match` functions like `keyMatch`, so I refer to the previous code specification to register `keyMatch4`, and only perform `ASSERT_EQ` assertion judgment for `keyGet*` functions.

Reply:
Golang Casbin doesn't introduce modules like `exprtk` to parse `keyMatch*` and `keyGet*` expressions for unit testing, but directly determines whether the return value of the function is equal to the expected result. 
At the same time, since the structure of [`ExprtkOtherFunction`](https://github.com/casbin/casbin-cpp/blob/master/include/casbin/model/exprtk_config.h#L110) designed in the previous code only supports the parsing of `*Match` functions like `keyMatch`, so I refer to the previous code specification to register `keyMatch4`, and only perform `ASSERT_EQ` assertion judgment for `keyGet*` functions.

Reply:
> At the same time, since the structure of [ExprtkOtherFunction](https://github.com/casbin/casbin-cpp/blob/master/include/casbin/model/exprtk_config.h#L110) designed in the previous code only supports the parsing of *Match functions like keyMatch, so I refer to the previous code specification to register keyMatch4, and only perform ASSERT_EQ assertion judgment for keyGet* functions.

But we need to use `keyGet*` in the matcher.😶

Reply:
> > At the same time, since the structure of [ExprtkOtherFunction](https://github.com/casbin/casbin-cpp/blob/master/include/casbin/model/exprtk_config.h#L110) designed in the previous code only supports the parsing of _Match functions like keyMatch, so I refer to the previous code specification to register keyMatch4, and only perform ASSERT_EQ assertion judgment for keyGet_ functions.
> 
> But we need to use `keyGet*` in the matcher.😶

I get what you mean. The `keyGet*` functions need to be registered for internal calls by the enforcer. I'll fix this next.